### PR TITLE
Add waiting_for_payment state and possible transitions

### PR DESCRIPTION
**Pull request by RobertCram**
PT Story: https://www.pivotaltracker.com/story/show/146016481

**Changes proposed in this pull request:**

1. Add a waiting_for_payment state and its possible transitions to the MembershipApplication

** Discussion **

1. Implementing it this way is no problem if the payment phase always comes AFTER the review phase. If payment is also possible during the review phase we should rethink this.

2. In the pivotal tracker story, the "received_payment" event would effect a transition from "waiting_for_payment" to "ready_for_review". That opened up the possibility of entering the "waiting_for_payment" state twice. I think a better option would be to either a) transition to "accepted" immediately, or -if absolutely needed-  b) to introduce a new intermediate state "ready_for_acceptance". I've implemented a).

3. At the moment it is possible to transition from "rejected" to "accepted". I'm not sure what the use case is for this, and therefore not sure how to include the "waiting_for_payment" state there. But it probably should be.

4. I don't speak Swedish, so I would appreciate it if someone who does could check my "translations".

Ready for review:
@thesuss @weedySeaDragon @patmbolger

### CODE REVIEW

**Comment by weedySeaDragon** [comment]
Just in case you haven't already seen these:

- [Membership Application states and transitions with notes (from discussions with the client) in `/docs`](https://github.com/AgileVentures/shf-project/blob/develop/docs/MembershipApplication%20States%20v1.1.pdf)

- [state diagram and table (generated from the code)  in the wiki](https://github.com/AgileVentures/shf-project/wiki/Membership-Application-State-Diagram-and-Transition-Table)

File: config/locales/sv.yml [hunk 1]
```diff
@@ -333,6 +346,10 @@ sv:
     cancel_waiting_for_applicant_btn: Ångra Behöver kompletteras
     waiting_for_applicant: *status_waiting_for_applicant
     applicant_updated_info: &event_applicant_updated_info Sökanden har kompletterat sin ansökan
+    ask_applicant_for_payment_btn: Behöver betalning
+    cancel_waiting_for_payment_btn: Ångra Behöver betalning
+    received_payment_btn: Fick betalning
```
Comment (thesuss) [inline config/locales/sv.yml]: Betalning inkommen

**Review by thesuss — APPROVED** [review]

**Comment by thesuss** [comment]
@RobertCram  I approved with a language comment.

**Review by weedySeaDragon — APPROVED** [review]
looks great!

**Pull request merged**